### PR TITLE
Fix empty pod ip pool type check

### PR DESCRIPTION
**Pull request by chenlin07**
**What this PR does / why we need it**:

Fix empty pod ip pool type check

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/cloudprovider/vsphereparavirtual/cloud.go [hunk 1]
```diff
@@ -112,12 +112,12 @@ func (cp *VSphereParavirtual) Initialize(clientBuilder cloudprovider.ControllerC
 	klog.V(0).Info("Initing vSphere Paravirtual Cloud Provider")
 
 	if vpcModeEnabled {
-		if podIPPoolType != PublicIPPoolType && podIPPoolType != PrivateIPPoolType {
-			klog.Fatalf("Pod IP Pool Type can be either Public or Private in VPC network, %s is not supported", podIPPoolType)
+		if podIPPoolType == "" {
+			klog.Fatalf("Pod IP Pool Type can't be empty in VPC network")
```
Comment (DanielXiao) [inline pkg/cloudprovider/vsphereparavirtual/cloud.go:116]: msg: --pod-ip-pool-type is required in the NSX-T VPC network

File: pkg/cloudprovider/vsphereparavirtual/cloud.go [hunk 2]
```diff
@@ -112,12 +112,12 @@ func (cp *VSphereParavirtual) Initialize(clientBuilder cloudprovider.ControllerC
...
 		}
 
-		if podIPPoolType == "" {
-			podIPPoolType = PrivateIPPoolType
+		if podIPPoolType != PublicIPPoolType && podIPPoolType != PrivateIPPoolType {
+			klog.Fatalf("Pod IP Pool Type can be either Public or Private in VPC network, %s is not supported", podIPPoolType)
```
Comment (DanielXiao) [inline pkg/cloudprovider/vsphereparavirtual/cloud.go:120]: msg: --pod-ip-pool-type can be either Public or Private in the NSX-T VPC network, %s is not supported

File: pkg/cloudprovider/vsphereparavirtual/cloud.go (hunk 1 above)
Comment (chenlin07) [reply pkg/cloudprovider/vsphereparavirtual/cloud.go]: fixed

File: pkg/cloudprovider/vsphereparavirtual/cloud.go [hunk 3]
```diff
@@ -173,7 +157,7 @@ func (cp *VSphereParavirtual) Initialize(clientBuilder cloudprovider.ControllerC
 	if RouteEnabled {
 		klog.V(0).Info("Starting routable pod controllers")
 
-		if err := routablepod.StartControllers(kcfg, client, cp.informMgr, ClusterName, clusterNS, ownerRef, vpcModeEnabled, podIPPoolType); err != nil {
+		if err := routablepod.StartControllers(kcfg, client, cp.informMgr, ClusterName, clusterNS, ownerRef, vpcModeEnabled, ippType); err != nil {
```
Comment (XudongLiuHarold) [inline pkg/cloudprovider/vsphereparavirtual/cloud.go:160]: change to `ipPoolType`?

File: pkg/cloudprovider/vsphereparavirtual/config.go [hunk 4]
```diff
@@ -135,3 +139,22 @@ func getRestConfig(svConfigPath string) (*rest.Config, error) {
...
+			return podIPPoolType, errors.New("--pod-ip-pool-type can be either Public or Private in NSX-T VPC network, " + podIPPoolType + " is not supported")
+
+		}
+	} else {
+		// NSX-T T1 or VDS network
+		if podIPPoolType != "" {
+			return podIPPoolType, errors.New("--pod-ip-pool-type can be set only when the network is VPC")
+		}
+	}
+	return podIPPoolType, nil
```
Comment (XudongLiuHarold) [inline pkg/cloudprovider/vsphereparavirtual/config.go:159]: No need to return `podIPPoolType` since you only valid not modify it

**Review by XudongLiuHarold — COMMENTED** [review]
generally looks good to me. some minor comments.

File: pkg/cloudprovider/vsphereparavirtual/cloud.go [hunk 5]
```diff
@@ -111,19 +105,9 @@ func newVSphereParavirtual(cfg *cpcfg.Config) (*VSphereParavirtual, error) {
...
-			podIPPoolType = PrivateIPPoolType
-		}
-	} else {
-		// NSX-T T1 or VDS network
-		if podIPPoolType != "" {
-			klog.Fatal("Pod IP Pool Type can be set only when the network is VPC")
-		}
+	ippType, err := checkPodIPPoolType(vpcModeEnabled, podIPPoolType)
+	if err != nil {
+		klog.Fatalf("Failed to check pod ip pool type: %v", err)
```
Comment (XudongLiuHarold) [inline pkg/cloudprovider/vsphereparavirtual/cloud.go:110]: `Invalid IP pool type: %v"

**Review by XudongLiuHarold — APPROVED** [review]
/lgtm

/approve

**Pull request merged**